handlers/azure_queue_messages_api_v1/handler/init.rb

```diff
@@ -0,0 +1,164 @@
+# Require the dependencies file to load the vendor libraries
+require File.expand_path(File.join(File.dirname(__FILE__), "dependencies"))
+
+class AzureQueueMessagesApiV1
+  # ==== Parameters
+  # * +input+ - The String of Xml that was built by evaluating the node.xml handler template.
+  def initialize(input)
+    # Set the input document attribute
+    @input_document = REXML::Document.new(input)
+
+    # Retrieve all of the handler info values and store them in a hash variable named @info_values.
+    @info_values = {}
+    REXML::XPath.each(@input_document, "/handler/infos/info") do |item|
+      @info_values[item.attributes["name"]] = item.text.to_s.strip
+    end
+
+    # Retrieve all of the handler parameters and store them in a hash variable named @parameters.
+    @parameters = {}
+    REXML::XPath.each(@input_document, "/handler/parameters/parameter") do |item|
+      @parameters[item.attributes["name"]] = item.text.to_s.strip
+    end
+
+    @debug_logging_enabled = ["yes","true"].include?(@info_values['enable_debug_logging'].downcase)
+    @error_handling = @parameters["error_handling"]
+
+    @tenent_id = @info_values["tenent_id"]
+    @client_id = @info_values["client_id"]
+    @client_secret = @info_values["client_secret"]
+    
+    @headers = @parameters["headers"].to_s.empty? ? {} : JSON.parse(@parameters["headers"])
+    @body = @parameters["body"]
+    @method = (@parameters["method"] || :get).downcase.to_sym
+    @storage_account = @parameters["storage_account"]
+    puts @parameters
+    @path = @parameters["path"]
+    @path = "/#{@path}" if !@path.start_with?("/")
+
+    @accept = 'text/xml'
+    @content_type = 'text/xml'
+  end
+
+  def execute
+    # Initialize return data
+    error_message = nil
+    error_key = nil
+    response_code = nil
+    retries = 0
+
+    begin
+      auth_url = "https://login.microsoftonline.com/#{@tenent_id}/oauth2/v2.0/token"
+      asset_url = "https://#{@storage_account}.queue.core.windows.net#{@path}"
+      puts "AUTH URL: #{auth_url}" if @debug_logging_enabled
+      puts "ASSET URL: #{asset_url}" if @debug_logging_enabled
+      puts "BODY: #{@body}" if @debug_logging_enabled
+
+      # Get Auth token
+      auth_response = RestClient::Request.execute \
+        method: 'POST', \
+        url: auth_url, \
+        payload: {
+          "client_id" => @client_id,
+          "client_secret" => @client_secret,
+          "grant_type" => "client_credentials",
+          "scope" => "https://#{@storage_account}.queue.core.windows.net/.default"
+        }, \
+        headers: {:content_type => 'application/x-www-form-urlencoded', :accept => @accept}
+       
+      # Parse token from request
+      puts "Access token retrieved." if @debug_logging_enabled
+      token = JSON.parse(auth_response.body)['access_token']
```

Comment:
I don't understand why you're parsing the response body as JSON when telling the server you only accept XML. I assume the server doesn't respect the accept header and only returns JSON. Probably best just to remove the accept header from the request.

Reply:
Removed